src/modules/perl.rs

```diff
@@ -0,0 +1,102 @@
+use super::{Context, Module, RootModuleConfig, SegmentConfig};
+
+use crate::configs::perl::PerlConfig;
+use crate::utils;
+
+/// Creates a module with the current perl version
+///
+/// Will display the perl version if any of the following criteria are met:
+///     - Current directory contains a `.pl` or a `.pm` file
+///     - Current directory contains a "Makefile.PL", "cpanfile", "META.json", "META.yml", or ".perl-version" file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_perl_project = context
+        .try_begin_scan()?
+        .set_files(&[
+            "Makefile.PL",
+            "cpanfile",
+            "META.json",
+            "META.yml",
+            ".perl-version",
+        ])
+        .set_extensions(&["pl", "pm"])
+        .is_match();
+
+    if !is_perl_project {
+        return None;
+    }
+
+    let perl_version = utils::exec_cmd("perl", &["-e", "print $^V;"])?.stdout;
+
+    let mut module = context.new_module("perl");
+    let config: PerlConfig = PerlConfig::try_load(module.config);
+    module.set_style(config.style);
+
+    module.create_segment("symbol", &config.symbol);
+    module.create_segment("version", &SegmentConfig::new(&perl_version));
+
+    Some(module)
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::modules::utils::test::render_module;
+    use ansi_term::Color;
+    use std::fs::File;
+    use std::io;
+
+    #[test]
+    fn folder_without_perl_files() -> io::Result<()> {
+        let dir = tempfile::tempdir()?;
+
+        let actual = render_module("perl", dir.path(), None);
+
+        let expected = None;
+        assert_eq!(expected, actual);
+        dir.close()
+    }
+
+    #[test]
+    fn folder_with_cpanfile_file() -> io::Result<()> {
+        let dir = tempfile::tempdir()?;
+        File::create(dir.path().join("cpanfile"))?.sync_all()?;
+
+        let actual = render_module("perl", dir.path(), None);
+
+        let expected = Some(format!(
+            "via {} ",
+            Color::Fixed(149).bold().paint("🐪 v5.26.1")
```

Comment:
The version needs to match with the output of the mock command.

Reply:
yes but test env all have different versions of perl. Linux 5.26.1, macos 5.30.2 and windows 5.30.3

Reply:
The problem is that the command you are running when getting the version in the module is different from the command in the mock function. Which means that the command is run directly and the output of the mock command is not used.

Reply:
ok I think i fixed that in last commit

Reply:
Checks are still failing. You need to replace the mock command with `perl -e print $^V;`.